src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -92,7 +96,12 @@ type StatefulSetResource struct {
 	schemaRegistryClientCertSecretKey  types.NamespacedName
 	serviceAccountName                 string
 	configuratorSettings               ConfiguratorSettings
-	logger                             logr.Logger
+	// hash of configmap containing configuration for redpanda, it's injected to
+	// annotation to ensure the pods get restarted when configuration changes
+	// this has to be retrieved lazily to achieve the correct order of resources
+	// being applied
+	configMapHashGetter func() (string, error)
```

Comment:
NIT: This could be named `configMapHash`, per Golang's naming conventions for accessors.

Reply:
client-go also uses the `Getter` pattern `BuildConfigFromKubeconfigGetter` - but I am fine with both approaches, whatever people prefer